src/main/java/chess/Piece.java

```diff
@@ -31,11 +32,37 @@ public void setSymbol(final Symbol symbol) {
     public void setStatus(final Status status) {
         this.status = status;
     }
-
-    public void setPosition(final Position position) {
-        this.position = position;
+    public void setPosition(final Position source) {
+        this.position = source;
+    }
+    public void setChessBoard(final ChessBoard chessBoard) {
+        this.chessBoard = chessBoard;
     }
 
+    public boolean move(final Position position, final ChessBoard chess) {
+        if (isPossibleMove(position, chess)) {
+           if (chess.getBoard()[this.position.getColumn()][this.position.getRow()] != null) {
+               chess.getBoard()[this.position.getColumn()][this.position.getRow()].die();
+           }
+           //chess.getBoard()[this.position.getColumn()][this.position.getRow()] = null;
+           chess.move2(this.position, null);
+           this.position = position;
+           chess.getBoard()[this.position.getColumn()][this.position.getRow()] = this;
+           return true;
+       }
+       return false;
+    }
+    public boolean isPossibleMove(final Position target, final ChessBoard chess) {
```

Comment:
This method also doesn't need pass the chessBoard

Reply:
Change made can you review it? please